src/main/java/gregtech/common/metatileentities/multi/electric/MetaTileEntityPowerSubstation.java

```diff
@@ -0,0 +1,638 @@
+                        ? MetaTileEntities.MAINTENANCE_HATCH
+                        : MetaBlocks.METAL_CASING.getState(MetalCasingType.PALLADIUM_SUBSTATION),
+                        EnumFacing.SOUTH);
+
+        GregTechAPI.PSS_BATTERIES.entrySet().stream()
+                // filter out empty batteries in example structures, though they are still
+                // allowed in the predicate (so you can see them on right-click)
+                .filter(entry -> entry.getValue().getCapacity() > 0)
+                .sorted(Comparator.comparingInt(entry -> entry.getValue().getTier()))
+                .forEach(entry -> shapeInfo.add(builder.where('B', entry.getKey()).build()));
+
+        return shapeInfo;
+    }
+
+    protected IBlockState getCasingState() {
+        return MetaBlocks.METAL_CASING.getState(BlockMetalCasing.MetalCasingType.PALLADIUM_SUBSTATION);
+    }
+
+    protected IBlockState getGlassState() {
+        return MetaBlocks.TRANSPARENT_CASING.getState(BlockGlassCasing.CasingType.LAMINATED_GLASS);
+    }
+
+    protected static final Supplier<TraceabilityPredicate> BATTERY_PREDICATE = () -> new TraceabilityPredicate(blockWorldState -> {
+        IBlockState state = blockWorldState.getBlockState();
+        if (GregTechAPI.PSS_BATTERIES.containsKey(state)) {
+            IBatteryData battery = GregTechAPI.PSS_BATTERIES.get(state);
+            // Allow unfilled batteries in the structure, but do not add them to match context.
+            // This lets you use empty batteries as "filler slots" for convenience if desired.
+            if (battery.getTier() != -1 && battery.getCapacity() > 0) {
+                String key = String.format("%s%s", PMC_BATTERY_HEADER, battery.getName());
+                BatteryMatchWrapper wrapper = blockWorldState.getMatchContext().get(key);
+                if (wrapper == null) wrapper = new BatteryMatchWrapper(battery);
+                blockWorldState.getMatchContext().set(key, wrapper.increment());
+            }
+            return true;
+        }
+        return false;
+    }, () -> GregTechAPI.PSS_BATTERIES.entrySet().stream()
+            .sorted(Comparator.comparingInt(entry -> entry.getValue().getTier()))
+            .map(entry -> new BlockInfo(entry.getKey(), null))
+            .toArray(BlockInfo[]::new))
+            .addTooltips("gregtech.multiblock.pattern.error.batteries");
+
+    @Override
+    public ICubeRenderer getBaseTexture(IMultiblockPart sourcePart) {
+        return Textures.PALLADIUM_SUBSTATION_CASING;
+    }
+
+    @NotNull
+    @Override
+    protected ICubeRenderer getFrontOverlay() {
+        return Textures.POWER_SUBSTATION_OVERLAY;
+    }
+
+    @Override
+    public void renderMetaTileEntity(CCRenderState renderState, Matrix4 translation, IVertexOperation[] pipeline) {
+        super.renderMetaTileEntity(renderState, translation, pipeline);
+        getFrontOverlay().renderOrientedState(renderState, translation, pipeline, getFrontFacing(), this.isActive(), true);
```

Comment:
This should use `this.isWorkingEnabled()` instead of `true` for the last parameter.